src/language-js/print/flow.js

```diff
@@ -328,13 +329,46 @@ function printFlow(path, options, print) {
     case "MatchLiteralPattern":
     case "MatchUnaryPattern":
     case "MatchIdentifierPattern":
+    case "MatchInstancePattern":
+    case "MatchInstanceObjectPattern":
     case "MatchMemberPattern":
     case "MatchBindingPattern":
     case "MatchObjectPattern":
     case "MatchObjectPatternProperty":
     case "MatchRestPattern":
     case "MatchArrayPattern":
       return printMatchPattern(path, options, print);
+
+    case "RecordExpression":
+      return [
+        print("recordConstructor"),
+        print("typeArguments"),
+        " ",
+        print("properties"),
+      ];
+    case "RecordExpressionProperties":
+      return printObject(path, options, print);
```

Comment:
Out of curiosity, why `properties` not directly on `RecordExpression`? It won't be a problem, and easy for us to locate `{`, but curious.